rocAL/rocAL/source/augmentations/geometry_augmentations/node_resize_mirror_normalize.cpp

```diff
@@ -0,0 +1,139 @@
+    for (uint i=0; i < _batch_size; i++ ) {
+        _mean_vx[3*i] = _mean[0];
+        _mean_vx[3*i+1] = _mean[1];
+        _mean_vx[3*i+2] = _mean[2];
+
+        _std_dev_vx[3*i] = _std_dev[0];
+        _std_dev_vx[3*i+1] = _std_dev[1];
+        _std_dev_vx[3*i+2] = _std_dev[2];
+    }
+    _mean_array = vxCreateArray(vxGetContext((vx_reference)_graph->get()), VX_TYPE_FLOAT32, _batch_size*3);
+    _std_dev_array = vxCreateArray(vxGetContext((vx_reference)_graph->get()), VX_TYPE_FLOAT32, _batch_size*3);
+    vx_status mean_status = VX_SUCCESS;
+    mean_status |= vxAddArrayItems(_mean_array,_batch_size*3, _mean_vx.data(), sizeof(vx_float32));
+    mean_status |= vxAddArrayItems(_std_dev_array,_batch_size*3, _std_dev_vx.data(), sizeof(vx_float32));
+    _mirror.create_array(_graph ,VX_TYPE_UINT32, _batch_size);
+    if(mean_status != 0)
+        THROW(" vxAddArrayItems failed in the resize mirror normalize node (vxExtrppNode_ResizeMirrorNormalizeCropbatchPD    )  node: "+ TOSTR(mean_status) + "  "+ TOSTR(mean_status))
+    
+    unsigned int chnShift = 0;
+    vx_scalar  chnToggle = vxCreateScalar(vxGetContext((vx_reference)_graph->get()),VX_TYPE_UINT32,&chnShift);
+
+    std::vector<uint32_t> dst_roi_width(_batch_size,_outputs[0]->info().width());
+    std::vector<uint32_t> dst_roi_height(_batch_size, _outputs[0]->info().height_single());
+
+    _dst_roi_width = vxCreateArray(vxGetContext((vx_reference)_graph->get()), VX_TYPE_UINT32, _batch_size);
+    _dst_roi_height = vxCreateArray(vxGetContext((vx_reference)_graph->get()), VX_TYPE_UINT32, _batch_size);
+
+    vx_status width_status, height_status;
+
+    width_status = vxAddArrayItems(_dst_roi_width, _batch_size, dst_roi_width.data(), sizeof(vx_uint32));
+    height_status = vxAddArrayItems(_dst_roi_height, _batch_size, dst_roi_height.data(), sizeof(vx_uint32));
+     if(width_status != 0 || height_status != 0)
+        THROW(" vxAddArrayItems failed in the resize mirror normalize node (vxExtrppNode_ResizeMirrorNormalizeCropbatchPD ) node: "+ TOSTR(width_status) + "  "+ TOSTR(height_status))
+
+   _node = vxExtrppNode_ResizeMirrorNormalizebatchPD(_graph->get(), _inputs[0]->handle(), _src_roi_width, _src_roi_height, _outputs[0]->handle(),
+                                                    _dst_roi_width, _dst_roi_height, _mean_array, _std_dev_array,
+                                                    _mirror.default_array() , chnToggle , _batch_size);
+
+    vx_status status;
+    if((status = vxGetStatus((vx_reference)_node)) != VX_SUCCESS)
+        THROW("Adding the resize mirror normalize node (vxExtrppNode_ResizeMirrorNormalizeCropbatchPD) node failed: "+ TOSTR(status))
+}
+
+void ResizeMirrorNormalizeNode::update_node()
+{
+    std::vector<uint32_t> src_roi_width, src_roi_height;
+    src_roi_width = _inputs[0]->info().get_roi_width_vec();
+    src_roi_height = _inputs[0]->info().get_roi_height_vec();
+
+    for(uint i = 0; i < _batch_size; i++)
+    {
+        int min_size = 800;
```

Comment:
How did we arrive at this min size?
Please add a comment stating if this is needed explicitly for MaskRCNN